### ISSUE TITLE
[Phase 4] Model Improvements (Optional)

### ISSUE THREAD

**Issue by Sakeeb91** [Sakeeb91/exoplanet-detection-ml#5]
# Phase 4: Model Improvements (Optional)

**Parent Issue:** #1

## Objective

If baseline RF is insufficient (<0.85 AUC), implement deep learning on raw light curves.

## Dependencies

- Phase 3 completed
- Baseline AUC < 0.85 (trigger condition)

### CNN Architecture
- [ ] Implement `src/models/cnn.py`
- [ ] Design 1D CNN with attention mechanism
- [ ] Add global max pooling for variable-length sequences
- [ ] Implement residual connections

### Data Loading
- [ ] Implement `src/models/data_loader.py`
- [ ] Create PyTorch Dataset class
- [ ] Add data augmentation (phase shifts, noise injection)
- [ ] Handle variable-length light curves (padding/truncation)

### Training
- [ ] Implement CNN training loop
- [ ] Add learning rate scheduling (cosine annealing)
- [ ] Early stopping based on validation loss
- [ ] Checkpoint best model

### Interpretability
- [ ] Implement Grad-CAM for 1D convolutions
- [ ] Visualize what CNN focuses on (should highlight transits)
- [ ] Compare attention maps with known transit locations

### CNN Architecture
```python
class TransitCNN(nn.Module):
    def __init__(self, seq_len: int = 3197):
        super().__init__()

        self.conv1 = nn.Conv1d(1, 32, kernel_size=5, padding=2)
        self.conv2 = nn.Conv1d(32, 64, kernel_size=5, padding=2)
        self.conv3 = nn.Conv1d(64, 128, kernel_size=5, padding=2)

        self.pool = nn.MaxPool1d(2)
        self.global_pool = nn.AdaptiveMaxPool1d(1)

        self.fc = nn.Sequential(
            nn.Linear(128, 64),
            nn.ReLU(),
            nn.Dropout(0.5),
            nn.Linear(64, 1),
            nn.Sigmoid()
        )

    def forward(self, x):
        x = F.relu(self.pool(self.conv1(x)))
        x = F.relu(self.pool(self.conv2(x)))
        x = F.relu(self.pool(self.conv3(x)))
        x = self.global_pool(x).squeeze(-1)
        return self.fc(x)
```

## Verification

```bash
python src/models/train_cnn.py --epochs 50
# Should complete in <2 hours on CPU
# Expected: AUC improvement of >0.02 over RF
```

## Definition of Done

- [ ] CNN outperforms RF by >0.02 AUC
- [ ] Training completes in <2 hours on CPU
- [ ] Grad-CAM visualizations show transit focus
- [ ] Model and training curves saved

## Labels

`phase-4`, `deep-learning`, `priority-medium`, `optional`
